### PR TITLE
fix: skip generating error events for unauthorized or forbidden errors

### LINKED ISSUE #91: test: random test fail with MockEventSQLDao
Screenshot 2024-08-01 at 21 52 13

### PR DESCRIPTION
**Pull request by duyhungtnn**
part of https://github.com/bucketeer-io/bucketeer/issues/1284

also fix #91

### CODE REVIEW

File: BucketeerTests/Mock/MockEventSQLDao.swift [hunk 1]
```diff
@@ -24,6 +25,9 @@ final class MockEventSQLDao: EventSQLDao {
     }
 
     func add(events: [Event]) throws {
+        lock.lock()
```
Comment (duyhungtnn) [inline BucketeerTests/Mock/MockEventSQLDao.swift:28]: fix: race condition that cause the unit test fail https://github.com/bucketeer-io/ios-client-sdk/issues/91

**Comment by duyhungtnn** [comment]
@cre8ivejp please help me to take a look

File: Bucketeer/Sources/Internal/Event/EventInteractor.swift [hunk 2]
```diff
@@ -168,6 +168,19 @@ final class EventInteractorImpl: EventInteractor {
     }
 
     func trackFetchEvaluationsFailure(featureTag: String, error: BKTError) throws {
+
+        if case .forbidden = error {
+            logger?.warn(message: "An forbidden error occurred. Please check your API Key.")
```
Comment (cre8ivejp) [inline Bucketeer/Sources/Internal/Event/EventInteractor.swift:181]: Can you merge the warning and error log?
What will happen if the register event API request fails?  Does it also log the errors?

Comment (duyhungtnn) [reply Bucketeer/Sources/Internal/Event/EventInteractor.swift:181]: unfortunately, the logger interface doesn't support passing the error as an input parameter like the Android SDK. I don't want to introduce breaking changes on the logger.

Comment (duyhungtnn) [reply Bucketeer/Sources/Internal/Event/EventInteractor.swift:173]: > What will happen if the register event API request fails? Does it also log the errors?

we did nothing when it failed. The events still on the database and waiting for the next synchronization

Comment (duyhungtnn) [reply Bucketeer/Sources/Internal/Event/EventInteractor.swift:181]: I updated it, we will not generate a log when the register event fails.
https://github.com/bucketeer-io/ios-client-sdk/pull/95/commits/fbcf080ec36aba557d2e2b35edb71ee35f15c17e

**Review by cre8ivejp — APPROVED** [review]
Thank you!
Nice work!

**Pull request merged**